units/src/amount.rs

```diff
@@ -864,6 +864,22 @@ impl Amount {
     /// Returns [None] if overflow occurred.
     pub fn checked_rem(self, rhs: u64) -> Option<Amount> { self.0.checked_rem(rhs).map(Amount) }
 
+    /// Unchecked addition.
+    /// This is the unsafe variant of addition.  Overflow instead of panic.
+    pub fn unchecked_add(self, rhs: Amount) -> Amount {
+        Self(self.0 + rhs.0)
+    }
+
+    /// Unchecked subtraction.
+    /// This is the unsafe variant of subtraction.  Overflow instead of panic.
+    pub fn unchecked_sub(self, rhs: Amount) -> Amount {
```

Comment:
And this is not "unchecked", at least in the sense it is usually used in Rust.

 I'd consider calling it `raw_sub`.

Reply:
Unchecked doesn't imply potential UB, I think. We have `unsafe` for that.

Reply:
`unchecked` is definitely used here in the normal sense of "this is the version of the function that doesn't do checks, and will do something wrong given wrong input".

`undefined` I will accept we shouldn't use, since it's potentially ambiguous whether *we're* not defining it or whether the *language* is not defining it. Although again, you need the `unsafe` keyword for the latter, so it's not really ambiguous.

Reply:
> you need the `unsafe` keyword for the latter, so it's not really ambiguous.

You're not wrong but I also expect people to open issues asking why it's not `unsafe`.

Reply:
> version of the function that doesn't do checks

I get what you're saying, but in stdlib `unchecked_` implies `unsafe`, [including `u64`  method named exactly like the same](https://doc.rust-lang.org/std/primitive.u64.html#method.unchecked_add) and they are actually unchecked, not just "unchecked in release build, but checked in debug build". So it will be at least slightly confusing. :D

Reply:
The ones in `std` are not really unchecked but C-like - UB on overflow. That's quite different. Actually I'm going to complain that the name is misleading. And IIRC I've seen `_unchecked` method somewhere which wasn't UB.

Reply:
Yes, I'm sure we've had this discussion before but I can't find it. But we use `_unchecked` in multiple places in this project, and in rust-bech32, and in an old version of byteorder, and in tokio.

Reply:
`sum` comes from `std` trait so it cannot be renamed.